### ISSUE TITLE
Visual bug component chooser

### ISSUE THREAD

**Issue by wilzbach** [CyberShadow/bugzilla-meta#4]
![image](https://user-images.githubusercontent.com/4370550/37940833-2b78f8be-316b-11e8-9e46-8d1aaf42ff94.png)

I really like this mask, maybe we can keep it?
We should be able to find generic icons like the toolbox for tools etc.

Maybe we should also include dub in the list, but clicking on dub's icon would forward the user to dub's issue tracker?

**Comment by CyberShadow** [comment]
Seems a bit like a gimmick in our case. Mozilla has different bug reporting workflows for the products above.

We would also need to think up of / draw icons for DRuntime, Phobos, dlang.org...

**Comment by CyberShadow** [comment]
> Maybe we should also include dub in the list, but clicking on dub's icon would forward the user to dub's issue tracker?

Yes, I have a note to check if we can add "fake" components that just redirect the user to the component's bug tracker. There's a bunch of things we can do this for.

**Comment by wilzbach** [comment]
> We would also need to think up of / draw icons for DRuntime, Phobos, dlang.org...

dlang.org could be the typical www globe - DRuntime <-> Phobos would be more difficult.
Though maybe just a stylistic "R" and "P" could work.

> Seems a bit like a gimmick in our case. 

Not only, but yes mostly. Though the description text should be a lot more helpful to the user to where they should report the bug too. I find a wrongly inserted bug report almost every week.

**Comment by CyberShadow** [comment]
So, there's a whole bunch of ways to enter bugs in Bugzilla / BMO.

# 1. Default bug entry form

This is what we have on issues.dlang.org currently.

![](https://dump.thecybershadow.net/7d5df1c5b0b726385b33d5c9967bc888/Screen%20Shot%202018-04-03%20at%2013.08.51.png)

# 2. Default bug entry form + advanced fields

Clicking a button on the default bug entry form reveals some additional fields.

![](https://dump.thecybershadow.net/e2e41ff6481fbc339f0da4b338ddde95/Screen%20Shot%202018-04-03%20at%2013.08.56.png)

# 3. GuidedBugEntry

This method is provided by [the GuidedBugEntry extension](https://github.com/mozilla-bteam/bmo/tree/master/extensions/GuidedBugEntry). It is BMO-specific, and the list of products (and their icons) is hard-coded, so we would need to patch it in our fork.

![](https://dump.thecybershadow.net/6db289b2459efb2965dd11beb73ad334/Screen%20Shot%202018-04-03%20at%2012.53.30.png)

![](https://dump.thecybershadow.net/9d743da4978022815a02defc9964df82/Screen%20Shot%202018-04-03%20at%2013.03.56.png)

![](https://dump.thecybershadow.net/0be394f0d83a13f6fa636fdab8c5dfa1/Screen%20Shot%202018-04-03%20at%2013.04.18.png)

# 4. format=guided

Apparently there is a new guided bug entry method in the Bugzilla core (i.e. not an extension), which presents the field as a linear, long form:

![](https://dump.thecybershadow.net/36a22af9f4693af5de1dd4dc6854dd20/Screen%20Shot%202018-04-03%20at%2013.17.57.png)

This is still nicer than the default one as it doesn't overwhelm the user, and allows them to go through the fields one at a time.

# 5. Custom forms

Bugzilla allows defining custom forms for reporting bugs via its configuration:

![](https://dump.thecybershadow.net/c5483dff7ee57171ef549c7074585d43/Screen%20Shot%202018-04-03%20at%2013.21.24.png)

These don't have much interactivity, but they do provide a form with fields for every bit of information that the user is expected to provide.
